src/applications/disability-benefits/686c-674/config/chapters/report-add-a-spouse/does-live-with-spouse/doesLiveWithSpouse.js

```diff
@@ -47,46 +32,24 @@ export const uiSchema = {
     'ui:errorMessages': { required: 'Please give a brief explanation' },
   },
   currentSpouseAddress: {
-    'ui:title': 'Your spouse’s address',
-    'ui:options': {
-      expandUnder: 'spouseDoesLiveWithVeteran',
-      expandUnderCondition: false,
-    },
-    currentSpouseCountry: {
-      'ui:title': 'Country',
-      'ui:required': doesLiveTogether,
-      'ui:errorMessages': { required: 'Please select a country' },
-    },
-    currentSpouseStreet: {
-      'ui:title': 'Street',
-      'ui:required': doesLiveTogether,
-      'ui:errorMessages': { required: 'Please enter a street address' },
-    },
-    currentSpouseLine2: {
-      'ui:title': 'Line 2',
-    },
-    currentSpouseLine3: {
-      'ui:title': 'Line 3',
-    },
-    currentSpouseCity: {
-      'ui:title': 'City',
-      'ui:required': doesLiveTogether,
-      'ui:errorMessages': { required: 'Please enter a city' },
-    },
-    currentSpouseState: {
-      'ui:title': 'State',
-      'ui:required': doesLiveTogether,
-      'ui:errorMessages': {
-        required: 'Please enter a state, or country if outside the U.S.',
-      },
-    },
-    currentSpousePostalCode: {
-      'ui:required': doesLiveTogether,
+    ...addressUISchema(false, 'currentSpouseAddress', doesLiveTogether),
+    ...{
+      'ui:title': 'Your spouse’s address',
       'ui:options': {
-        widgetClassNames: 'usa-input-medium',
+        expandUnder: 'spouseDoesLiveWithVeteran',
+        expandUnderCondition: false,
+        updateSchema: (formData, formSchema) => {
+          // if someone selects a country, and then changes their mind and selects 'no' for spouseDoesLiveWithVeteran,
+          // The collapsed form will silently throw an error because some fields are required based on country.
+          // manually clearning the required array fixes this issue.
+          if (formData.spouseDoesLiveWithVeteran) {
+            return {
+              required: [],
```

Comment:
The form will silently throw a validation error on some of the address fields if a user returns to the very first question of this page and selects `yes, my spouse lives with me`, causing the address fields to collapse. Adding a top-level `updateSchema` to the address itself and clearing the `required` fields array fixes this issue.

Reply:
Sounds fun!